js/XAPI.js

```diff
@@ -1124,14 +1123,16 @@ class XAPI extends Backbone.Model {
       state
     });
 
-    // Pass the new state to the LRS.
-    this.xapiWrapper.sendState(activityId, actor, collectionName, registration, newState, null, null, (error, xhr) => {
-      if (error) {
-        Adapt.trigger('xapi:lrs:sendState:error', error);
-      }
+    // Pass the new state to the LRS, but debounced to limit the server updates to periodic updates
+    _.debounce(() => {
```

Comment:
The debounce function returns a new function each time, the new function is a debounced version of the original function passed as the first argument, the first argument is not executed at any point. You've used `_.debounce` here as if it were `_.defer`.

You'll need to split `sendState` into two named functions. Debounce the `sendStateToServer` function in `initialize` as you had before - by reassigning the instance function as its debounced version: 

`this.sendStateToServer = _.debounce(this.sendStateToServer.bind(this), 500);`. 

Then call `this.sendStateToServer()` at the end of `sendState`.